synapse/handlers/auth.py

```diff
@@ -1950,7 +1950,7 @@ def run(*args: Tuple, **kwargs: Dict) -> Awaitable:
         # need to use a tuple here for ("password",) not a list since lists aren't hashable
         auth_checkers[(LoginType.PASSWORD, ("password",))] = check_password
 
-    api.register_password_auth_provider_callbacks(hooks, auth_checkers=auth_checkers)
+    api.register_password_auth_provider_callbacks(auth_checkers=auth_checkers, **hooks)
```

Comment:
I'm actually finding this code really hard to reason about. It seems like the abstract `hooks` variable saves like 2 lines of code? Can we just be explicit?

Reply:
We probably can. FWIW I suggested we do it this way for consistency with how we load legacy spam checkers and third-party rule modules, but maybe it's just only relevant there.

Reply:
Stripped this out now and inlined it. I think it's better this way, though maybe it was justifiable with the sheer number of hooks for the spam checker modules.

Reply:
Yeah that was my point, the third-party rules and spam checker modules have enough callbacks that it probably makes it worth it, but that's not really the case here.

Reply:
Definitely fair, but when we come back to this code later we probably won't be looking at these all at once. Thanks for updating it!

Reply:
Yeah I definitely agree that it's better without a dict here, I was just trying to provide additional context.